qxf2/qxf2-lambdas
# Lambda to sync graphql neo4j

In this PR I have added a lambda function that updates the neo4j database with respect to Graphql database.
It checks if all the employees in Graphql are present in neo4j. If not it adds that employee to the neo4j database
It also compares the status of the employees between Graphql and neo4j and updates the status in neo4j database.

The lambda has been deployed here https://ap-south-1.console.aws.amazon.com/lambda/home?region=ap-south-1#/functions/sync_neo4j_and_graphql

The Lamda has been set to work with survey test instance as of now.

How to test:
1. Go to the lambda https://ap-south-1.console.aws.amazon.com/lambda/home?region=ap-south-1#/functions/sync_neo4j_and_graphql?tab=code
2. Run test
- You should notice the response 
`
[
  "No new employees to add",
  [
    "The active status of all the employees are up to date"
  ]
]
`
3. Now go to the survey-backend database instance http://15.206.119.69:7474/
4. Login to database instance with credentials given in Trello card `Test-Survey credentials`
5. Run command
`
MATCH (e:Employees)
WHERE e.email="ajitava.deb@qxf2.com"
DETACH DELETE e
`
6. This command will delete employee Ajitava from database
7. Now go back to the lambda and run test again
8. Check the execution result, You should notice a new response that says Ajitava has been added to database
9. Now go to the survey-backend database instance http://15.206.119.69:7474
10. Run the command
`
MATCH (m:Employees) WHERE m.email="ajitava.deb@qxf2.com" return m
`
11. You should notice the employee Ajitava has been added again to the database
12. Now again in the database instance http://15.206.119.69:7474, run the command
`
MATCH (e:Employees)
WHERE e.email="ajitava.deb@qxf2.com"
set e.status="N"
`
13. This will set the active status of employee to N
14. Now go to the lambda and run the test again
15. Check the execution result, you will notice the result saying the employee status of employee ajitava has been updated


Files changed:
sync_neo4j_and_graphql/requirements.txt
sync_neo4j_and_graphql/sync_neo4j_graphql.py
